### PR TITLE
build: moving docs back to hybrid site to support redirects

### PR DESCRIPTION
**Pull request by zchenwei**
*Issue #, if available:*
#1111 

Some links on docs site that require redirects do not work as expect because we are deploying our docs site with SSG.

*Description of changes:*

Moving docs site back to SSR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by reesscot** [comment]
@zchenwei I don't think docs changes should have a changeset since we don't publish `docs` as a package.

**Review by reesscot — CHANGES_REQUESTED** [review]
Please check the PR preview, it doesn't seem to be working right now.

**Comment by zchenwei** [comment]
> g



> Please check the PR preview, it doesn't seem to be working right now.

It should be working now. It was stuck last Friday, the build succeeded though

**Review by dbanksdesign — APPROVED** [review]
LGTM! :shipit:

File: docs/next.config.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ module.exports = withNextPluginPreval({
   },
 
   // These redirects are because of the IA change from previous docs
-  redirects() {
+  async redirects() {
```
Comment (wlee221) [inline docs/next.config.js:54]: For my understanding, why `async`?

Comment (zchenwei) [reply docs/next.config.js:54]: `Next` config requires it to be async anyway and we didn't configure it correctly in the first place :( <- https://nextjs.org/docs/api-reference/next.config.js/redirects. But yea, `Next` doesn't explain much about why it should be async.

**Review by reesscot — APPROVED** [review]
LGTM!

**Pull request merged**